### PR TITLE
Lint: timeslider

### PR DESCRIPTION
**Pull request by HMarzban**
Lint `/src/static/js/timeslider.js`

### CODE REVIEW

File: src/static/js/timeslider.js [hunk 1]
```diff
@@ -28,11 +30,81 @@ const Cookies = require('./pad_utils').Cookies;
 const randomString = require('./pad_utils').randomString;
 const hooks = require('./pluginfw/hooks');
 
-let token, padId, export_links;
+let token, padId, exportLinks, socket, changesetLoader, BroadcastSlider;
```
Comment (rhansen) [inline src/static/js/timeslider.js:33]: If I recall correctly, the `socket` and `BroadcastSlider` vars need to be on the `globalThis` object (`window`) because other `.js` files (and maybe code in the HTML?) reference them. Rather than scope them to this file, use `window.socket` and `window.BroadcastSlider` to ensure that these lint fixes don't change behavior.

I realize that it is gross to do `window.foo`, but cleaning that up is out of scope for a lint fix PR.

File: src/static/js/timeslider.js [hunk 2]
```diff
@@ -44,14 +116,14 @@ function init() {
 
     // ensure we have a token
     token = Cookies.get('token');
-    if (token == null) {
+    if (!token) {
```
Comment (rhansen) [inline src/static/js/timeslider.js:119]: Undo this change; it's not a lint fix.

File: src/static/js/timeslider.js [hunk 3]
```diff
@@ -28,11 +30,81 @@ const Cookies = require('./pad_utils').Cookies;
 const randomString = require('./pad_utils').randomString;
 const hooks = require('./pluginfw/hooks');
 
-let token, padId, export_links;
+let token, padId, exportLinks, socket, changesetLoader, BroadcastSlider;
+const fireWhenAllScriptsAreLoaded = [];
```
Comment (rhansen) [inline src/static/js/timeslider.js:34]: Moving the code around makes it difficult to review changes. Would you move these back to the order they were originally in? (That might not be possible if the original order throws `ReferenceError`.) ESLint will complain about `no-use-before-define`, but those are warnings not errors.

File: src/static/js/timeslider.js [hunk 4]
```diff
@@ -71,90 +143,35 @@ function init() {
...
+    exportLinks = $('#export > .exportlink');
 
     $('button#forcereconnect').click(() => {
       window.location.reload();
     });
 
-    exports.socket = socket; // make the socket available
-    exports.BroadcastSlider = BroadcastSlider; // Make the slider available
+    // exports.socket = socket;
+    // exports.BroadcastSlider = BroadcastSlider;
```
Comment (rhansen) [inline src/static/js/timeslider.js:163]: Please don't introduce commented-out code.

**Review by rhansen — COMMENTED** [review]
Please rebase onto `develop`.

File: src/static/js/timeslider.js (hunk 4 above)
Comment (JohnMcLear) [reply src/static/js/timeslider.js:163]: Weird that I did...  I wonder why..  Thanks for spotting ...  Will investigate why/how this happened because I know not to.

File: src/static/js/timeslider.js (hunk 3 above)
Comment (JohnMcLear) [reply src/static/js/timeslider.js:34]: afaik this was to fix an error (probably not ``initialized``), I didn't address lint warnings..

File: src/static/js/timeslider.js (hunk 4 above)
Comment (JohnMcLear) [reply src/static/js/timeslider.js:163]: Oh...  This wasn't my PR :)  I'm an idjit 📦
